traversal/selector/exploreRecursive_test.go

```diff
@@ -0,0 +1,243 @@
+		parentsSelector := ExploreAll{recursiveEdge}
+		sideSelector := ExploreAll{recursiveEdge}
+		subTree := ExploreFields{map[string]Selector{
+			"Parents": parentsSelector,
+			"Side":    ExploreRecursive{sideSelector, sideSelector, maxDepth},
+		}, []PathSegment{
+			PathSegmentString{S: "Parents"},
+			PathSegmentString{S: "Side"},
+		},
+		}
+		s := ExploreRecursive{subTree, subTree, maxDepth}
+		n := fnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+			mb.Insert(knb.CreateString("Parents"), vnb.CreateList(func(lb fluent.ListBuilder, vnb fluent.NodeBuilder) {
+				lb.Append(vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+					mb.Insert(knb.CreateString("Parents"), vnb.CreateList(func(lb fluent.ListBuilder, vnb fluent.NodeBuilder) {}))
+					mb.Insert(knb.CreateString("Side"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+						mb.Insert(knb.CreateString("cheese"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+							mb.Insert(knb.CreateString("whiz"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {}))
+						}))
+					}))
+				}))
+			}))
+			mb.Insert(knb.CreateString("Side"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+				mb.Insert(knb.CreateString("real"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+					mb.Insert(knb.CreateString("apple"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
+						mb.Insert(knb.CreateString("sauce"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {}))
+					}))
+				}))
+			}))
+		})
+		rn := n
+		rs = s
+		rs = rs.Explore(rn, PathSegmentString{S: "Parents"})
+		rn, err = rn.TraverseField("Parents")
+		Wish(t, rs, ShouldEqual, ExploreRecursive{subTree, parentsSelector, maxDepth})
+		Wish(t, err, ShouldEqual, nil)
+		rs = rs.Explore(rn, PathSegmentInt{I: 0})
+		rn, err = rn.TraverseIndex(0)
+		Wish(t, rs, ShouldEqual, ExploreRecursive{subTree, subTree, maxDepth - 1})
+		Wish(t, err, ShouldEqual, nil)
+		rs = rs.Explore(rn, PathSegmentString{S: "Parents"})
+		rn, err = rn.TraverseField("Parents")
+		Wish(t, rs, ShouldEqual, ExploreRecursive{subTree, parentsSelector, maxDepth - 1})
+		Wish(t, err, ShouldEqual, nil)
+		rn = n
+		rs = s
+		rs = rs.Explore(rn, PathSegmentString{S: "Side"})
+		rn, err = rn.TraverseField("Side")
+		Wish(t, rs, ShouldEqual, ExploreRecursive{subTree, ExploreRecursive{sideSelector, sideSelector, maxDepth}, maxDepth})
+		Wish(t, err, ShouldEqual, nil)
+		rs = rs.Explore(rn, PathSegmentString{S: "real"})
+		rn, err = rn.TraverseField("real")
+		Wish(t, rs, ShouldEqual, ExploreRecursive{subTree, ExploreRecursive{sideSelector, sideSelector, maxDepth - 1}, maxDepth})
```

Comment:
Got a little unclear to me what these cases are trying to focus in on.  Might make sense but my eyes just kinda crossed.

Maybe with these fixtures doing some stuff with JSON (just in heredocs?  no need to bite off a whole fixture file loading thing right way) would be even more of a payoff than the earlier instances.